### ISSUE TITLE
Power Tools

### ISSUE THREAD

**Issue by mikeoswego** [rocky-linux/rocky-tools#50]
I have a CentOS 8.4 box that failed the migration due to no repo for the package libedit-devel.
I had edited /etc/yum.repos.d/CentOS-Linux-PowerTools.repo to enable the repo, libedit-devel is part of that.

I'm very sorry that I lost the log file with the exact wording in my rush to recover the system.

This was the second box I tried to migrate, the first one migrated perfectly (it didn't have the PowerTools repo enabled.)

Thanks for you work on this project!

**Comment by pajamian** [comment]
I recently saw this issue with someone else, the problem, if it is the same one (I'm only guessing without the log) is this:

1.  You enable epel, or some other 3rd-party repository and want to install something from it.
2. dnf install pkgname returns an error because it depends on a package in appstream, which you have disabled.
3. dnf --enablerepo=appstream install pkgname works (but it notably doesn't leave appstream enabled by default).
4. Some time later you run migrate2rocky, which gets to the distro-sync stage and then fails.

The distro-sync fails because in the meantime the dependency that is in appstream got updated after which the package you installed from epel also got updated and now depends on the newer package in appstream.  The distro-sync tries to install the newer package from epel, but it can't because appstream is disabled and it can't get the newer dependency from appstream.

I should note that if you had tried to run `dnf update` before running migrate2rocky it will also have failed for the same reason.  If dnf update won't run then there is little chance that migrate2rocky will complete.

I am considering putting a dnf update at the beginning of migrate2rocky to address this issue.  If it fails then the rest of teh script won't run and your system won't be left in an unstable state.  If it succeeds then your system will be in a better position to continue the migration under.

**Comment by mikeoswego** [comment]
Actually, 'dnf update' was run just before attempting the migration, it has always run fine after enabling the powertools repo.
Again, and  maybe this was my error, I did not do 'dnf --enablerepo=powertools', I manually edited /etc/yum.repos.d/CentOS-Linux-PowerTools.repo to enable it.  Then libedit-devel was installed as a pre-requisite for Asterisk.

I never went back and changed that, there would have been no reason for me to do that.  However, I see that the backup of the /etc/yum.repos.d/CentOS-Linux-PowerTools.repo at the time of the migration had the enable line changed back to "enable=0"  Is this the likely cause and if I make sure the line reads "enable=1" things should go well?

I will try again and be sure to capture the log if needed.
